### PR TITLE
Feat/forms

### PR DESCRIPTION
**Pull request by henask12**
### Description
This pull request addresses the following project requirements:

1. **Current User Method**: 
- [x] Created a new method called `current_user` in the `ApplicationController` to provide access to the current user data.
- [x] this method retrieves the first user from the database.
- [x] Authentication is not implemented at this stage, and it will be added in future projects.

2. **User Interaction Forms**:
- [x] Created forms to enable the following user interactions:
- [x]  Creating a new Post on behalf of the current user using the `current_user` method defined in `ApplicationController`.
- [x] Creating a comment on behalf of the current user, utilizing the `current_user` method.
3. **Likes for Posts**:
- [x]  Implemented the ability for users to add likes to Posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+Rails.application.routes.draw do
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+
+  # Defines the root path route ("/")
+  # root "articles#index"
+
+  root "users#index"
+
+  resources :users, only: [:show, :index] do
+    resources :posts, only: [:show, :index, :new, :create] do
```
Comment (FrederickMih) [inline config/routes.rb:10]: - [ ] For this Milestone, we are requested to use only two pair of routes(:new and :create) corresponding to the `#new` and `#create` posts controller actions
Screenshot from 2023-10-06 21-57-28

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -7,3 +7,6 @@
     <p>There's no users, please</p>
   <% end %>
 </section>
+<section class="add-section flex-column">
+  <%= link_to "Add a new post", new_user_post_path(@users) %>
```
Comment (FrederickMih) [inline app/views/users/index.html.erb:11]: - [ ] Hi @henask12, though you added forms for creating posts, the error below on the screen record prevents a user from creating a post. Hence a user can't equally create a comment. Let's once more kindly double check and fix :wrench: the issue.

[Screencast from 10-06-2023 09:42:07 PM.webm](https://github.com/henask12/Blogify/assets/63087491/4c94571c-23f1-4d51-b0fd-12d5fb71c163)

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @henask12, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Both posts and comments are created on behalf of the currently logged in user as requested :rocket: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Alfredbis29 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi @henask12 
Your project is complete! There is nothing else to say other than... Great job clap Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Your project is professional :white_check_mark:
 * Good Job :+1:
 
 Now you can merge this branch and move on 1st_place_medal: HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :


 
 ## Optional suggestions

- [x] N/A

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**